asreview/project.py

```diff
@@ -0,0 +1,659 @@
+                kwargs_copy["mode"]))
+
+        # update project file
+        project_fp = Path(self.project_path, PATH_PROJECT_CONFIG)
+
+        with open(project_fp, "r") as f:
+            config = json.load(f)
+
+        config.update(kwargs_copy)
+
+        with open(project_fp, "w") as f:
+            json.dump(config, f)
+
+        self._config = config
+        return config
+
+    def rename(self, project_name_new):
+        """Rename a project id.
+
+        This function only works for projects in ASReview LAB  web interface.
+        This is the result of the file storage in
+        asreview.webapp.utils.project_path.asreview_path.
+
+        Arguments
+        ---------
+        project_id: str
+            The current project_id.
+        project_name_new: str
+            The new project name to be converted into a new
+            project_id.
+
+        Returns
+        -------
+        str:
+            The new project_id.
+        """
+
+        # create a new project_id from project name
+        project_id_new = _create_project_id(project_name_new)
+        project_path_new = Path(asreview_path(), project_id_new)
+
+        if (self.project_path == project_path_new):
+            # nothing to do
+            return self
+
+        if (self.project_path !=
+                project_path_new) & is_project(project_path_new):
+            raise ValueError(f"Project '{project_path_new}' already exists.")
+
+        self.project_path.rename(project_path_new)
+        self.project_path = project_path_new
+        self.project_id = project_id_new
+
+        # update the project file
+        config = self.config
+
+        config["id"] = project_id_new
+        config["name"] = project_name_new
+
+        self.config = config
+        self._config = config
+
+        return self
+
+    def add_dataset(self, file_name):
+        """Add file path to the project file.
+
+        Add file to data subfolder and fill the pool of iteration 0.
+        """
+        self.update_config(dataset_path=file_name)
+
+        # fill the pool of the first iteration
+        as_data = read_data(self.project_path)
+
+        with open_state(self.project_path, read_only=False) as state:
+
+            # save the record ids in the state file
+            state.add_record_table(as_data.record_ids)
+
+            # if the data contains labels, add them to the state file
+            if as_data.labels is not None:
+
+                labeled_indices = np.where(as_data.labels != LABEL_NA)[0]
+                labels = as_data.labels[labeled_indices].tolist()
+                labeled_record_ids = as_data.record_ids[
+                    labeled_indices].tolist()
+
+                # add the labels as prior data
+                state.add_labeling_data(
+                    record_ids=labeled_record_ids,
+                    labels=labels,
+                    notes=[None for _ in labeled_record_ids],
+                    prior=True)
```

Comment:
The labels in the dataset should not be added to the state file if the mode is simulation.